### PR TITLE
[NME] Set Block

### PR DESCRIPTION
**Pull request by onekit-boss**
an emty block,  for "a = b"

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15090/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15090/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15090/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15090/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15090/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15090/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15090/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15090/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15090/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15090/merge/testResults/webgl2playwright/index.html

**Comment by onekit-boss** [comment]
@deltakosh

File: packages/tools/nodeEditor/src/blockTools.ts [hunk 1]
```diff
@@ -355,6 +355,11 @@ export class BlockTools {
                 triangleWaveBlock.kind = WaveBlockKind.Triangle;
                 return triangleWaveBlock;
             }
+            case "SetBlock": {
+                const cosBlock = new TrigonometryBlock("Set");
```
Comment (Popov72) [inline packages/tools/nodeEditor/src/blockTools.ts:359]: nit: `setBlock` would be better than `cosBlock`

File: packages/tools/nodeEditor/src/components/nodeList/nodeListComponent.tsx [hunk 2]
```diff
@@ -110,7 +110,8 @@ export class NodeListComponent extends React.Component<INodeListComponentProps,
         ArcSinBlock: "Outputs the inverse of the sine value based on the input value",
         ArcTan2Block: "Outputs the inverse of the tangent value based on the input value",
         ArcTanBlock: "Outputs the inverse of the tangent value based on the input value",
-        CosBlock: "Outputs the cosine value based on the input value",
+        CosBlock: "Outputs the alias based on the input value",
+        SesBlock: "Outputs the cosine value based on the input value",
```
Comment (Popov72) [inline packages/tools/nodeEditor/src/components/nodeList/nodeListComponent.tsx:114]: The description is reversed, and `SesBlock` should be `SetBlock`

**Review by Popov72 — COMMENTED** [review]
I'm not sure what the block is for, though?

**Review by deltakosh — COMMENTED** [review]
What are the use cases?

**Comment by onekit-boss** [comment]
for example:
float red = 0.3;
float green = red;
float blue = green;

**Comment by onekit-boss** [comment]
[SetBlock_Demo.json](https://github.com/BabylonJS/Babylon.js/files/15283473/SetBlock_Demo.json)

File: packages/tools/nodeEditor/src/blockTools.ts (hunk 1 above)
Comment (onekit-boss) [reply packages/tools/nodeEditor/src/blockTools.ts:359]: Okay, I has fixed it.

Comment (onekit-boss) [reply packages/tools/nodeEditor/src/blockTools.ts:359]: next week, I will create a to Tool  to convert GLSL code to NME, so we need the "SetBlock".

File: packages/tools/nodeEditor/src/components/nodeList/nodeListComponent.tsx (hunk 2 above)
Comment (onekit-boss) [reply packages/tools/nodeEditor/src/components/nodeList/nodeListComponent.tsx]: Sometimes, we need the alias of the Block to do something.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15090/merge/testResults/webgpuplaywright/index.html

**Comment by onekit-boss** [comment]
@deltakosh

**Comment by deltakosh** [comment]
Well I'm not super convinced but the code change is minimal

**Review by deltakosh — APPROVED** [review]

**Pull request merged**